recipes/msgpack/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class MsgpackConan(ConanFile):
+    name = "msgpack"
+    description = "The official C++ library for MessagePack"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/msgpack/msgpack-c"
+    topics = ("conan", "msgpack", "message-pack", "serialization")
+    license = "BSL-1.0"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "build_type", "compiler"
+    options = {"fPIC": [True, False], "shared": [True, False], "header_only": [True, False], "with_boost": [True, False]}
+    default_options = {"fPIC": True, "shared": False, "header_only": False, "with_boost": False}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.header_only:
+            self.settings.clear()
+            del self.options.shared
+            del self.options.fPIC
+            del self.options.with_boost
+
+    def requirements(self):
+        if not self.options.header_only and self.options.with_boost:
+            self.requires.add("boost/1.69.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "msgpack-c-cpp-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
```

Comment:
```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
```
**+ more modification below + add a **`_autotools=None`** statement at the top**
Cache the cmake object in the object, because right now cmake might be run twice: in `build` and in `package`.

Reply:
done